rpc/gnmi/decimal64-deprecation.md

```diff
@@ -0,0 +1,168 @@
+transmission of the data.  As stated above, the source of the original data is
+likely a low-bit-count ADC which can easily be encoded as a single 32bit binary
+floating point number.  The message used in gNMI for Decimal64 uses two integer
+values, one 32 bit the other 64 bit for a total of 96 bits, and creates a nested
+structure which itself consumes additional space.  Given that most language
+implementations of protocol buffers represent message types using pointers and
+most systems in which gNMI will be deployed are likely 64-bit, this is an
+additional 64-bit minimum for a total of 160 bits, or 5X overhead in storage.
+
+Furthermore, because the Decimal64 type is a complex message, there is
+additional computational overhead in the protocol buffer encoding and decoding
+as compared to the native protocol buffer float type. A simple Go benchmark
+(below) shows a 60% increase in protocol buffer encoding overhead.  This does
+not include the previously mentioned application level encoding and decoding
+between the original float and a Decimal64 message, which does not exist when a
+native float type is used directly.
+
+On an Intel Xeon W-2135 @ 3.7GHz
+
+|                      | Double      | Decimal64   |
+| BenchmarkEncode+Rand | 432.8 ns/op | 615.5 ns/op |
+| BenchmarkRand        | 13.2 ns/op  | 12.6  ns/op |
+| BenchmarkEncode only | 419.6 ns/op | 602.9 ns/op |
+
+```go
+package decimal64
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+
+	pb "github.com/openconfig/gnmi/proto/gnmi/gnmi_go_proto"
+)
+
+func BenchmarkDoubleVal(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		f := rand.Float64()
+		t := &pb.TypedValue{Value: &pb.TypedValue_DoubleVal{DoubleVal: f}}
+		_, err := proto.Marshal(t)
+		if err != nil {
+			b.Fatalf("marshal error: %v", err)
+		}
+	}
+}
+
+func BenchmarkRandFloat64(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		rand.Float64()
+	}
+}
+
+func BenchmarkDecimal64Val(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		df := rand.Int63()
+		t := &pb.TypedValue{Value: &pb.TypedValue_DecimalVal{&pb.Decimal64{Digits: df, Precision: 5}}}
+		_, err := proto.Marshal(t)
+		if err != nil {
+			b.Fatalf("marshal error: %v", err)
+		}
+	}
+}
+
+func BenchmarkRandInt63(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		rand.Int63()
+	}
+}
+```
+
+#### Confusing both implementers and operators
+
+Another significant inefficiency of the use of Decimal64 is one of inter-
+organizational human communication. The gNMI specification and Protocol Buffer
+definition include two ways of rendering fractional values, Decimal64 and
+double_val.  Because of this, we have some vendors using one or the other or both
+methods for various paths.  Downstream clients need to handle both cases and
+convert Decimal64 to float where it appears.  There is often the question of,
+“What matches the official OpenConfig schema”.  The goal of this proposal is to
+codify that answer as double_val for gNMI.
```

Comment:
ACK, I suggest that at this point this isn't really a proposal any more, since we're basically saying that we're doing it. Will make suggestions for how to reflect that.